node/src/reactor.rs

```diff
@@ -551,7 +577,22 @@ where
 
         // Dispatch the event, then execute the resulting effect.
         let start = self.clock.start();
-        let effects = self.reactor.dispatch_event(effect_builder, rng, event);
+
+        let (effects, keep_going) = if let Some(ctrl_ann) = event.as_control() {
+            // We've received a control event, which will _not_ be handled by the reactor.
+            match ctrl_ann {
+                ControlAnnouncement::FatalError { file, line, msg } => {
```

Comment:
I wonder whether it should be called `ShutdownAnnouncement` or `AbortAnnouncement` instead, since it always makes us exit?

(And whether it should be a wrapper type instead, _containing_ an event, so that `dispatch_event` doesn't need an unhandled `match` arm? Not sure how ugly that would be.)

Reply:
This is actually in preparation for having more than just the `FatalError` as a control announcement in the future - regular exits or some other data shall also be part of this.

Since events targeting the runner/runtime need to be handled in a special manner, grouping them inside a single enum ensure this extra logic only has to be implemented once.